File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    rank = {v:i for i,v in enumerate(players)}
    
    for call in callings:
        curRank = rank[call]
        
        rank[call] -= 1
        rank[players[curRank-1]] += 1
        
        players[curRank-1], players[curRank] = call, players[curRank-1]
        
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    let rank = {}
    players.map((v,i) => rank[v] = i)
    
    for (let call of callings){
        let curI = rank[call];
        let revI = curI - 1;

        rank[call] -= 1; 
        rank[players[revI]] += 1;

        players[curI] = players[revI];
        players[revI] = call;
    }
    return players;
}